utils/azureCredentialsManager.ts

```diff
@@ -0,0 +1,138 @@
+    }
+
+    public async getRegistries(subscription?: SubscriptionModels.Subscription, resourceGroup?: string, sortFunction?): Promise<ContainerModels.Registry[]> {
+        let registries: ContainerModels.Registry[] = [];
+
+        if (subscription && resourceGroup) {
+            //Get all registries under one resourcegroup
+            const client = this.getContainerRegistryManagementClient(subscription);
+            registries = await client.registries.listByResourceGroup(resourceGroup);
+
+        } else if (subscription) {
+            //Get all registries under one subscription
+            const client = this.getContainerRegistryManagementClient(subscription);
+            registries = await client.registries.list();
+
+        } else {
+            //Get all registries for all subscriptions
+            const subs: SubscriptionModels.Subscription[] = this.getFilteredSubscriptionList();
+            const subPool = new AsyncPool(MAX_CONCURRENT_SUBSCRIPTON_REQUESTS);
+
+            for (let i = 0; i < subs.length; i++) {
+                subPool.addTask(async () => {
+                    const client = new ContainerRegistryManagementClient(this.getCredentialByTenantId(subs[i].tenantId), subs[i].subscriptionId);
+                    let subscriptionRegistries: ContainerModels.Registry[] = await client.registries.list();
+                    registries = registries.concat(subscriptionRegistries);
+                });
+            }
+            await subPool.runAll();
+        }
+
+        if (sortFunction) {
+            registries.sort(sortFunction);
+        }
+
+        return registries;
+    }
+
+    public async getResourceGroups(subscription?: SubscriptionModels.Subscription): Promise<ResourceGroup[]> {
+        if (subscription) {
+            const resourceClient = new ResourceManagementClient(this.getCredentialByTenantId(subscription.tenantId), subscription.subscriptionId);
+            return await resourceClient.resourceGroups.list();
+        }
+        const subs = this.getFilteredSubscriptionList();
+        const subPool = new AsyncPool(MAX_CONCURRENT_SUBSCRIPTON_REQUESTS);
+        let resourceGroups: ResourceGroup[] = [];
+        //Acquire each subscription's data simultaneously
+        for (let i = 0; i < subs.length; i++) {
+            subscription = subs[i];
+            subPool.addTask(async () => {
+                const resourceClient = new ResourceManagementClient(this.getCredentialByTenantId(subscription.tenantId), subscription.subscriptionId);
+                const internalGroups = await resourceClient.resourceGroups.list();
+                resourceGroups = resourceGroups.concat(internalGroups);
+            });
+        }
+        await subPool.runAll();
+        return resourceGroups;
+    }
+
+    public getCredentialByTenantId(tenantId: string): ServiceClientCredentials {
+
+        const session = this.getAccount().sessions.find((s, i, array) => s.tenantId.toLowerCase() === tenantId.toLowerCase());
```

Comment:
nit: Let's use more verbose names: `s`->`azureSession`, `i`->`index` ? Also, do we need to declare unused `i` and `array` ?

Reply:
I'll get back to you on that, getCredentialsByTenantId was not implemented by me, rather it was a method existent in preexisting code from another class in the project (I think azureRegistryRootNode.ts). Ill take a look at it though.